tests/test_positioner.py

```diff
@@ -0,0 +1,112 @@
+from unittest.mock import Mock
+
+import pytest
+from ophyd import Component as Cpt, Signal
+from ophyd.sim import SynAxis, SignalRO
+
+from typhon.positioner import TyphonPositionerWidget
+
+from .conftest import show_widget
+
+
+class SimMotor(SynAxis):
+    low_limit_switch = Cpt(SignalRO, value=0)
+    high_limit_switch = Cpt(SignalRO, value=0)
+    low_limit = Cpt(Signal, value=-10)
+    high_limit = Cpt(Signal, value=10)
+    stop = Mock()
+
+
+@pytest.fixture(scope='function')
+def motor_widget(qtbot):
+    motor = SimMotor(name='test')
+    setwidget = TyphonPositionerWidget.from_device(motor)
+    qtbot.addWidget(setwidget)
+    yield motor, setwidget
+    if setwidget._status_thread and setwidget._status_thread.isRunning():
+        setwidget._status_thread.wait()
+
+
+def test_positioner_widget_no_limits(qtbot, motor):
+    setwidget = TyphonPositionerWidget.from_device(motor)
+    qtbot.addWidget(setwidget)
+    for widget in ('low_limit', 'low_limit_switch',
+                   'high_limit', 'high_limit_switch'):
+        assert getattr(setwidget.ui, widget).isHidden()
+
+
+def test_positioner_widget_fixed_limits(qtbot, motor):
+    motor.limits = (-10, 10)
+    widget = TyphonPositionerWidget.from_device(motor)
+    qtbot.addWidget(widget)
+    assert widget.ui.low_limit.text() == '-10'
+    assert widget.ui.high_limit.text() == '10'
+
+
+@show_widget
+def test_positioner_widget_with_signal_limits(motor_widget):
+    motor, widget = motor_widget
+    # Check limit switches
+    low_limit_chan = widget.ui.low_limit_switch.channel
+    assert motor.low_limit_switch.name in low_limit_chan
+    low_limit_chan = widget.ui.low_limit_switch.channel
```

Comment:
Copy-paste error?

Reply:
Luckily not masking another copy-paste error 😄 . Fixed in e1609a8